### PR TITLE
To support config dateformat for spy

### PR DESCRIPTION
**Pull request by yuebingc**
To support config dateformat for spy, e.g. %(currentTime)

### CODE REVIEW

**Comment by gavlyukovskiy** [comment]
Hi @yuebingc, thanks for the PR, please add tests for this configuration option and I can merge it.

**Comment by yuebingc** [comment]
Done, also fixed generic compiling issue in eclipse.

File: datasource-decorator-spring-boot-autoconfigure/src/main/java/com/github/gavlyukovskiy/boot/jdbc/decorator/flexypool/FlexyPoolDataSourceDecorator.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ public DataSource decorate(String beanName, DataSource dataSource) {
             if (customizers != null) {
                 customizers.forEach(customizer -> customizer.customize(beanName, configurationBuilder, dataSourceClass));
             }
-            FlexyPoolDataSource<DataSource> flexyPoolDataSource = new FlexyPoolDataSource<>(configurationBuilder.build(), connectionAcquiringStrategyFactories);
+            FlexyPoolDataSource<DataSource> flexyPoolDataSource = new FlexyPoolDataSource<DataSource>(configurationBuilder.build(), connectionAcquiringStrategyFactories);
```
Comment (gavlyukovskiy) [inline datasource-decorator-spring-boot-autoconfigure/src/main/java/com/github/gavlyukovskiy/boot/jdbc/decorator/flexypool/FlexyPoolDataSourceDecorator.java:79]: Sorry, but why is that necessary? What kind of eclipse warning do you get?
